internal/recorder/http_recorder.go

```diff
@@ -58,44 +56,57 @@ func New(t *testing.T) *Recorder {
 }
 
 func removeSensitiveDataFromRecordings(t *testing.T, recorderTransport *recorder.Recorder) {
-	allowedHeaders := map[string]bool{
-		"Content-Type": true,
-		"User-Agent":   true,
-	}
-
 	recorderTransport.AddHook(
 		func(i *cassette.Interaction) error {
-			for header := range i.Request.Headers {
-				if _, ok := allowedHeaders[header]; !ok {
-					delete(i.Request.Headers, header)
-				}
-			}
-			for header := range i.Response.Headers {
-				if _, ok := allowedHeaders[header]; !ok {
-					delete(i.Response.Headers, header)
-				}
-			}
+			skip429Response(i)
+			redactHeaders(i)
 
 			domain := os.Getenv("AUTH0_DOMAIN")
 			require.NotEmpty(t, domain, "removeSensitiveDataFromRecordings(): AUTH0_DOMAIN is empty")
 
 			redactSensitiveDataInClient(t, i, domain)
-
-			i.Request.Host = strings.Replace(i.Request.Host, domain, RecordingsDomain, -1)
-			i.Request.URL = strings.Replace(i.Request.URL, domain, RecordingsDomain, -1)
-
-			i.Response.Duration = time.Millisecond
```

Comment:
We don't really need to set the duration to 1ms as we already set `SkipRequestLatency: true` on the recorder options.